Mathlib/Init/Control/Combinators.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura
+-/
+import Mathlib.Mathport.Rename
+
+/-! ## Monad combinators, as in Haskell's Control.Monad. -/
+
+universe u v w
+
+def List.mmap {m : Type u → Type v} [Monad m] {α : Type w} {β : Type u} (f : α → m β) :
+    List α → m (List β)
+  | [] => return []
+  | h :: t => do
+    let h' ← f h
+    let t' ← List.mmap f t
+    return (h' :: t')
```

Comment:
The naming scheme has changed: instead of prefixing with an `m` for the monadic analogue, we now suffix with an `M`. So Lean4 already has `List.mapM`, etc.

Could you revise this to use those (with appropriate #align)?

Reply:
Will do

Reply:
How about `mjoin`, which I would really call `Monad.join`?

Reply:
similarly for `mcond`, `mwhen`?

Reply:
For now, I'll make it `joinM`, `condM`, `whenM`.